frontend/server/src/DAO/Problems.php

```diff
@@ -558,14 +558,14 @@ public static function getProblemsSolvedCount(\OmegaUp\DAO\VO\Identities $identi
     }
 
     /**
-     * @return list<array{alias: string, solved: bool, title: string, username: string}>
+     * @return list<array{alias: string, solved: int, title: string, username: string}>
      */
     public static function getProblemsByUsersInACourse(string $courseAlias) {
         $sql  = '
            SELECT
                 rp.alias,
                 rp.title,
-                rp.solved,
+                IF(ISNULL(rp.solved), FALSE, IF(rp.solved, TRUE, FALSE)) AS solved,
```

Comment:
```suggestion
                CAST(IFNULL(rp.solved, FALSE) AS TINYINT) AS solved,
```

ese `TINYINT` es esencial para que se detecte como `bool` en vez de `int`.

Reply:
si eso no funciona, se podrá hacer algo para que el `int` no escape de esta función y desde el punto de vista del que lo llama sólo vea `bool`?

Reply:
Ahí va otro intento

Reply:
aún manteniéndola como entero, esta expresión se puede simplificar un poco:

```suggestion
                IFNULL(rp.solved, FALSE) AS solved,
```

Reply:
Listo